src/dmd/mars.d

```diff
@@ -245,21 +246,37 @@ private int tryMain(size_t argc, const(char)** argv)
         return EXIT_SUCCESS;
     }
 
-    if (global.params.mcpuUsage)
+    /*
+    Prints a supplied usage text to the console and
+    returns the exit code for the help usage page.
+
+    Returns:
+        `EXIT_SUCCESS` if no errors occurred, `EXIT_FAILURE` otherwise
+    */
+    static byte printHelpUsage(string help)
     {
-        import dmd.cli : CLIUsage;
-        auto help = CLIUsage.mcpu;
         printf("%.*s", help.length, &help[0]);
-        return EXIT_SUCCESS;
+        return global.errors ? EXIT_FAILURE : EXIT_SUCCESS;
     }
 
-    if (global.params.transitionUsage)
+    /*
+    Check for each usage flag whether it has been toggled
+    If so, print its help page and exit
```

Comment:
It's returning a string, not printing anything nor exiting.

Reply:
It generates code that does this.